lib/core/src/sync/mod.rs

```diff
@@ -0,0 +1,49 @@
+pub(crate) mod model;
+
+use anyhow::Result;
+use async_trait::async_trait;
+
+use self::model::{sync::Record, SyncData};
+
+#[async_trait]
+pub trait SyncModule {
+    /// Connects to a gRPC stream
+    async fn connect(&self) -> Result<()>;
+
+    /// Applies the changes received from the stream to the local database
+    async fn apply_changes(&self, changes: &[Record]) -> Result<()>;
+
+    /// Retrieves the changes since a specified [record_index](Record::record_index)
+    async fn get_changes_since(&self, record_index: u64) -> Result<Vec<Record>>;
+
+    /// Adds a record to the remote
+    async fn set_record(&self, data: SyncData) -> Result<()>;
```

Comment:
Since we are using a list of Records as response to get_changes_since I think we should stick to that structure also on set_record. SyncData seems a higher abstraction.

Reply:
That's actually been changed in #509, see [here](https://github.com/breez/breez-sdk-liquid/pull/509/files#diff-39d1d030470dc22070380fe8f700011c8f39961bbbf0245cf825c4bffedc1b15R32). I opted for having `get_changes_since` also decode the data.

Reply:
@hydra-yse it is hard to review this PR by looking at changes in another PR.
I think either merge both into one PR or move the fixes that are related here to this PR.

Reply:
Sure, I'll merge the new interface definition here.

Reply:
The fact that we have a trait with set_record & get_changes_since means we have another implementation that will use these which will be triggered by the app. I think we need to see that implementation before we can determine if we need the interface to look like this.
I am OK with merging both PRs together, move to the implementation of the sync itself (at least partial) so we can review that effectively.